geoips/plugins/modules/output_checkers/geotiff.py

```diff
@@ -23,6 +23,76 @@
 name = "geotiff"
 
 
+def random_modification(input_path, output_path, max_modification=0.1):
+    """Randomly modify the file from input path, and output it to output_path."""
+    from osgeo import gdal
+    from osgeo import gdalconst
+    from numpy.random import rand
+
+    # Open the input GeoTIFF file
+    src_ds = gdal.Open(input_path, gdalconst.GA_ReadOnly)
+    driver = gdal.GetDriverByName("GTiff")
+    dst_ds = driver.CreateCopy(output_path, src_ds, 0)
+
+    # Randomly modify the temporary copy
+    band = dst_ds.GetRasterBand(1)
+    data = band.ReadAsArray().astype("float64")
+    data += (rand() - 0.5) * max_modification * 2
+    band.WriteArray(data)
+
+    # Close the dataset
+    dst_ds = None
+
+
+def get_test_files():
+    """Return a Series of GeoTIFF paths, of which are randomly modified from compare."""
+    from os import makedirs
+    from os.path import exists, join
+    import shutil
+    from importlib.resources import files
+
+    # Ensure the output directory exists
+    savedir = str(files("geoips") / "../../test_data/test_geotiffs/pytest") + "/"
```

Comment:
This resolves https://github.com/NRLMMD-GEOIPS/geoips/pull/379#discussion_r1371991996